rllib/algorithms/ppo/torch/ppo_torch_rl_module.py

```diff
@@ -155,26 +117,44 @@ def from_model_config(
         else:
             raise ValueError(f"Unsupported activation: {activation}")
 
+        obs_dim = observation_space.shape[0]
         fcnet_hiddens = model_config["fcnet_hiddens"]
         vf_share_layers = model_config["vf_share_layers"]
         free_log_std = model_config["free_log_std"]
         use_lstm = model_config["use_lstm"]
 
+        if vf_share_layers:
+            shared_encoder_config = FCConfig(
+                input_dim=obs_dim,
+                hidden_layers=fcnet_hiddens,
+                activation=activation,
+                output_dim=model_config["fcnet_hiddens"][-1],
+            )
+        else:
+            shared_encoder_config = IdentityConfig(output_dim=obs_dim)
+
         if use_lstm:
-            assert vf_share_layers, "LSTM not supported with vf_share_layers=False"
-            encoder_config = LSTMConfig(
+            pi_encoder_config = LSTMConfig(
+                input_dim=shared_encoder_config.output_dim,
```

Comment:
Why do we have these branches in the `RLModule`? If the user wants an `LSTMEncoder` or `shared_encoder`, does it not make sense for them to specify these in the config?

E.g.

```python
PPOConfig(shared_encoder=LSTMEncoderConfig(recurrent_size=16))
```

As this is currently written, we would need to add a `if use_cnn` block, a `if use_gnn` block, a `use_gru` block, etc for every `RLModule` we write, no?

Reply:
Absolutely, but this method only converts from the old model config dict for now.
Have a look at TorchPPORLmodule.setup() to see the code that instantiates from potential user configs.